apps/studio/components/interfaces/Storage/FilesBuckets.tsx

```diff
@@ -0,0 +1,182 @@
+import { useState } from 'react'
+import { useRouter } from 'next/router'
+
+import { useParams } from 'common'
+import { EmptyBucketState } from './EmptyBucketState'
+import { CreateBucketModal } from './CreateBucketModal'
+import { useBucketsQuery } from 'data/storage/buckets-query'
+import { Bucket } from 'data/storage/buckets-query'
+import { ScaffoldSection } from 'components/layouts/Scaffold'
+import { FormHeader } from 'components/ui/Forms/FormHeader'
+import {
+  Button,
+  Card,
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuSeparator,
+  DropdownMenuTrigger,
+} from 'ui'
+import { Plus, MoreVertical, Edit, Trash2, FolderOpen } from 'lucide-react'
+import { StoragePolicies } from './StoragePolicies/StoragePolicies'
+import { StorageSettings } from './StorageSettings/StorageSettings'
+import { S3Connection } from './StorageSettings/S3Connection'
+
+export const FilesBuckets = () => {
+  const { ref } = useParams()
+  const router = useRouter()
+  const [showCreateBucketModal, setShowCreateBucketModal] = useState(false)
+
+  const { data: buckets = [], isLoading } = useBucketsQuery({ projectRef: ref })
+
+  // Filter for standard buckets (files)
+  const filesBuckets = buckets.filter((bucket: Bucket) => bucket.type === 'STANDARD')
+
+  const handleCreateBucket = () => {
+    setShowCreateBucketModal(true)
+  }
+
+  // Determine which tab is active based on the current path
+  const getActiveTab = () => {
+    const path = router.asPath
+    if (path.includes('/storage/files/settings')) return 'settings'
+    if (path.includes('/storage/files/policies')) return 'policies'
+    return 'buckets'
+  }
+
+  const activeTab = getActiveTab()
+
+  if (isLoading) {
+    return (
+      <div className="flex items-center justify-center h-32">
+        <div className="animate-spin rounded-full h-8 w-8 border-b-2 border-foreground"></div>
+      </div>
+    )
+  }
+
+  // Render content based on active tab
+  const renderTabContent = () => {
+    switch (activeTab) {
+      case 'settings':
+        return (
+          <div className="space-y-12">
+            <StorageSettings />
+            <S3Connection />
+          </div>
+        )
+      case 'policies':
+        return <StoragePolicies />
+      case 'buckets':
+      default:
+        if (filesBuckets.length === 0) {
+          return (
+            <>
+              <EmptyBucketState bucketType="files" onCreateBucket={handleCreateBucket} />
```

Comment:
feedback (not sure if already planned down the road)

Just wondering cause the settings and policy pages won't be available without a bucket created, which will be hiding a couple of things that aren't bucket related like Settings -> S3 Connection, and Policies -> Policies on the storage schema directly

Reply:
Discussed on Slack with @burmecia:

> ...those S3 connection config are providing S3-compatible access, so they’re relevant to all bucket types including analytic and vector bucket type ...It doesn’t matter how many buckets are created, all the created buckets are using the same S3 connection info. Even with no bucket created, those info can still be there. 

So I need to extract S3 Connection as a separate sidebar item outside of bucket types. Sorry for this change!

Reply:
Following up about policies on same thread in Slack.